agent/consul/stream/event_publisher_test.go

```diff
@@ -56,6 +56,13 @@ func TestEventPublisher_SubscribeWithIndex0(t *testing.T) {
 		Payload: simplePayload{key: "sub-key", value: "the-published-event-payload"},
 	}
 	require.Equal(t, expected, next)
+
+	// Subscriber should not see events for other keys
+	publisher.Publish([]Event{{
+		Topic:   testTopic,
+		Payload: simplePayload{key: "other-key", value: "this-should-not-reach-the-subscriber"},
+	}})
+	assertNoResult(t, eventCh)
```

Comment:
The previous test that was removed had cases for all the permutations of namespace/name/partition etc. too. I don't think that necessary here provided we have an exhaustive test of the `TopicKey` matching method that is being used elsewhere. I'll assume we do but I haven't got to it yet!